Don't separate field/union/record childs of interface

sdroege opened the pull request:
... and also re-export all the field types from the crate root.

sdroege committed:
Don't separate field/union/record childs of interface

Their order relative to each other is relevant.

sdroege committed:
Re-export record/class/interface/union field type from the crate root

bilelmoussaoui commented on src/interface.rs:
```diff
@@ -31,6 +31,18 @@ impl Prerequisite {
 }
 
+#[derive(Clone, Debug, XmlDeserialize)]
+pub enum InterfaceField {
```
Can you fix the clippy lint?

sdroege replied:
`clippy::large_enum_variant` you mean? I don't think it's actually a problem here, so I'd just silence it if that's OK with you?

sdroege replied:
There's also

```
warning: this creates an owned instance just for comparison
  --> src/version.rs:80:9
   |
80 |         self.to_string() == other
   |         ^^^^^^^^^^^^^^^^ help: try: `self`
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.91.0/index.html#cmp_owned
   = note: `#[warn(clippy::cmp_owned)]` on by default
```

That one needs a bit more work though.

bilelmoussaoui replied:
> clippy::large_enum_variant you mean? I don't think it's actually a problem here, so I'd just silence it if that's OK with you?

yes just silence that one.



> There's also
> 
> ```
> warning: this creates an owned instance just for comparison
>   --> src/version.rs:80:9
>    |
> 80 |         self.to_string() == other
>    |         ^^^^^^^^^^^^^^^^ help: try: `self`
>    |
>    = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.91.0/index.html#cmp_owned
>    = note: `#[warn(clippy::cmp_owned)]` on by default
> ```
> 
> That one needs a bit more work though.

we can figure that another time

sdroege replied:
The reason why it's so huge is the `Type` and `AnyType` fields further down btw. Some strategic boxing here would probably be useful but `xmlserde` does not seem to support putting a field into a `Box`.

sdroege replied:
Anyway, ignored those for now.

sdroege committed:
Mark class/interface/record/union `Field` enums as `#[allow(clippy::large_enum_variant)]`

Because of the `Type` / `AnyType` fields they're quite huge and boxing
would probably be useful, but `xmlserde` does not seem to support that.
